subprojects/core/src/main/java/org/gradle/api/internal/artifacts/dsl/dependencies/DependencyAdderExtensionModule.java

```diff
@@ -0,0 +1,248 @@
+     */
+    public static void call(DependencyAdder self, CharSequence dependencyNotation, @ClosureParams(value = SimpleType.class, options = "org.gradle.api.artifacts.ExternalModuleDependency") Closure<?> configuration) {
+        self.add(dependencyNotation, ConfigureUtil.configureUsing(configuration));
+    }
+
+    /**
+     * Add a dependency.
+     *
+     * @param group the group
+     * @param name the name
+     * @param version the version
+     */
+    public static void call(DependencyAdder self, @Nullable String group, String name, @Nullable String version) {
+        self.add(group, name, version);
+    }
+
+    /**
+     * Add a dependency.
+     *
+     * @param group the group
+     * @param name the name
+     * @param version the version
+     * @param configuration an action to configure the dependency
+     */
+    public static void call(DependencyAdder self, @Nullable String group, String name, @Nullable String version, @ClosureParams(value = SimpleType.class, options = "org.gradle.api.artifacts.ExternalModuleDependency") Closure<?> configuration) {
+        self.add(group, name, version, ConfigureUtil.configureUsing(configuration));
+    }
+
+    /**
+     * Add a dependency.
+     *
+     * @param group the group
+     * @param name the name
+     * @param version the version
+     * @param classifier the classifier
+     * @param extension the extension
+     */
+    public static void call(DependencyAdder self, @Nullable String group, String name, @Nullable String version, @Nullable String classifier, @Nullable String extension) {
+        self.add(group, name, version, classifier, extension);
+    }
+
+    /**
+     * Add a dependency.
+     *
+     * @param group the group
+     * @param name the name
+     * @param version the version
+     * @param classifier the classifier
+     * @param extension the extension
+     * @param configuration an action to configure the dependency
+     */
+    public static void call(DependencyAdder self, @Nullable String group, String name, @Nullable String version, @Nullable String classifier, @Nullable String extension, @ClosureParams(value = SimpleType.class, options = "org.gradle.api.artifacts.ExternalModuleDependency") Closure<?> configuration) {
+        self.add(group, name, version, classifier, extension, ConfigureUtil.configureUsing(configuration));
+    }
+
+    /**
+     * Add a dependency. The map may contain the following keys:
+     * <ul>
+     *   <li>{@code group}</li>
+     *   <li>{@code name}</li>
+     *   <li>{@code version}</li>
+     *   <li>{@code classifier}</li>
+     *   <li>{@code extension}</li>
+     * </ul>
+     *
+     * @param map a map of configuration parameters for the dependency
+     */
+    public static void call(DependencyAdder self, Map<String, CharSequence> map) {
+        call(self, map, null);
+    }
+
+    private static final Set<String> LEGAL_MAP_KEYS = ImmutableSet.of("group", "name", "version", "classifier", "extension");
```

Comment:
Did you intentionally make this extension vs ext?  In the existing map notation, the key is 'ext'.

There's also a 'configuration' key, but I'm OK with dropping this because I think you can set it in ModuleDependency if you need to

Reply:
No, I didn't mean to change this.